osgp/platform/osgp-adapter-ws-shared/src/main/java/org/opensmartgridplatform/adapter/ws/infra/specifications/JpaDeviceSpecifications.java

```diff
@@ -29,61 +27,69 @@
-            throw new ArgumentNullOrEmptyException("organisation");
+            throw new ArgumentNullOrEmptyException(ORGANISATION);
         }
 
-        return new Specification<Device>() {
-            @Override
-            public Predicate toPredicate(final Root<Device> deviceRoot, final CriteriaQuery<?> query,
-                    final CriteriaBuilder cb) {
+        return ((deviceRoot, query, cb) -> this.createPredicateForOrganisation(deviceRoot, query, cb, organisation));
+    }
+
+    private Predicate createPredicateForOrganisation(final Root<Device> deviceRoot, final CriteriaQuery<?> query,
+            final CriteriaBuilder cb, final Organisation organisation) {
 
-                final Subquery<Long> subquery = query.subquery(Long.class);
-                final Root<DeviceAuthorization> deviceAuthorizationRoot = subquery.from(DeviceAuthorization.class);
-                subquery.select(deviceAuthorizationRoot.get("device").get("id").as(Long.class));
-                subquery.where(cb.equal(deviceAuthorizationRoot.get("organisation"), organisation.getId()));
+        final Subquery<Long> subquery = query.subquery(Long.class);
+        final Root<DeviceAuthorization> deviceAuthorizationRoot = subquery.from(DeviceAuthorization.class);
+        subquery.select(deviceAuthorizationRoot.get(DEVICE).get(ID).as(Long.class));
+        subquery.where(cb.equal(deviceAuthorizationRoot.get(ORGANISATION), organisation.getId()));
 
-                return cb.in(deviceRoot.get("id")).value(subquery);
-            }
-        };
+        return cb.in(deviceRoot.get(ID)).value(subquery);
     }
 
     @Override
     public Specification<Device> hasDeviceIdentification(final String deviceIdentification, final boolean exactMatch)
             throws ArgumentNullOrEmptyException {
         if (StringUtils.isEmpty(deviceIdentification)) {
-            throw new ArgumentNullOrEmptyException("deviceIdentification");
+            throw new ArgumentNullOrEmptyException(DEVICE_IDENTIFICATION);
         }
 
-        return new Specification<Device>() {
-            @Override
-            public Predicate toPredicate(final Root<Device> deviceRoot, final CriteriaQuery<?> query,
-                    final CriteriaBuilder cb) {
-                if (exactMatch) {
-                    return cb.equal(deviceRoot.<String> get("deviceIdentification"), deviceIdentification);
-                } else {
-                    return cb.like(cb.upper(deviceRoot.<String> get("deviceIdentification")),
-                            deviceIdentification.toUpperCase());
-                }
-            }
-        };
+        return ((deviceRoot, query, cb) -> this.createPredicateForDeviceIdentification(deviceRoot, query, cb,
+                deviceIdentification, exactMatch));
+    }
+
+    private Predicate createPredicateForDeviceIdentification(final Root<Device> deviceRoot,
+            final CriteriaQuery<?> query, final CriteriaBuilder cb, final String deviceIdentification,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "query". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS1172)